doc/user/layouts/partials/create-source/connector/kafka/details.html

```diff
@@ -9,6 +9,13 @@
   [SSL-encrypted](#ssl-encrypted-kafka-details) or
   [Kerberized Kafka clusters](#kerberized-kafka-details).
 
+#### Partition offsets
+
+The `start_offset` option comes with some quirks to be aware of:
+
+- If fewer offsets than partitions are provided, then the remaining partitions will start at offset 0. This is true if the user provides `start_offset=1` or `start_offset=[1, ...]`.
+- If more offsets than partitions are provided, then any partitions added later will incorrectly be read from that offset. So, if you have a single partition, but you provide `start_offset=[1,2]`, when you add the second partition you will miss the first 2 records of data.
+
```

Comment:
@JLDLaughlin @ruchirK - Can you both review this update? Unless I'm misunderstanding, you have conflicting takes on the second bullet here.

Reply:
I think that both bullets are accurate, although I don't think the behavior of "partitions added later will incorrectly be read from that offset" is incorrect per se it's just something that may be surprising to the user. I'm not sure what better behavior looks like - I suppose ideally materialize would check the number of partitions at source creation time and then complain if the number of start_offsets specified didn't match the number of partitions but this could get tricky.

Previously, the documentation had said something along the lines of "if you only specify one offset then that gets used for all partitions" which is no longer true and I don't think these docs conflict with that at all.